### DISCUSSION TITLE
Help in "Error querying SMP"

### DISCUSSION THREAD

**Discussion by Robcio35** [phax/phoss-smp#287] [Q&A]
Hello,
I just finished installing my PHOSS SMP - XML (peppol1.netfly.be) on a Linux server, with Tomcat 10.x as ROOT application.
I successfully registered the SMP in SML (SMK).
I created my first participant iso6523-actorid-upis::9925:netfly successfully.

But when I use the tool "Peppol Practical - Participant information", to get information about 9925:netfly, I get the following error :
_Error querying SMP.
Technical details: SMPClientException - Unknown error thrown by SMP server (Connect to http://B-199a7359d901c310a3c927c15b9a1806.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu:80 [B-199a7359d901c310a3c927c15b9a1806.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu/146.59.3.207] failed: Connect timed out)
Caused by: ConnectTimeoutException - Connect to http://B-199a7359d901c310a3c927c15b9a1806.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu:80 [B-199a7359d901c310a3c927c15b9a1806.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu/146.59.3.207] failed: Connect timed out_

I see nothing in catalina.out relative to this query, and finally I'm asking me what I configured incorrectly ?
- might be this error due to that I didn't define any endpoint yet for this participant ?
- or something at the server level ?

Any suggestion will be very helpful.

![image](https://github.com/user-attachments/assets/2d30efd5-326a-4915-b6d2-b39bac8c2cab)

**Comment by phax** [comment]
@Robcio35 i was trying to access http://peppol1.netfly.be/ from my machine, but I get a connection timeout. So my assumption is, that there is some inbound firewall/proxy that is not yet configured correctly. I would start searching there. hth

**Reply by Robcio35** [reply]
Thank you,
Probably you tried to access in the same time than I was trying to change configuration...
I put all parameters back as before any changes, and you can access now.

Other information :
I configured the SMP as ROOT on TOMCAT 10.x
I was able to register the SMP in SML.
Then I created a Participant group : 9925:netfly.
Finally I created 2 endpoints for this group.

Thanks

**Reply by Robcio35** [reply]
Addendum : 
I think that basically the problem is that my Tomcat forcibly redirect 80 port to 443 port...

**Comment by Robcio35** [comment]
Finally I solved my problem, simply allowing Tomcat to use HTTP and HTTPS !
R.

**Reply by phax** [reply]
Great that you were able to resolve it. If you have some links that you could share, where you found the missing piece of information, it might be helpful for others. Thanks :)

**Reply by Robcio35** [reply]
Of course @phax !

I changed 2 things in TOMCAT configuration.

**In /opt/tomcat/conf/server.xml**
I changed the connector section from this 
```
    <Connector port="80" protocol="HTTP/1.1"
               connectionTimeout="20000"
               redirectPort="443"
               maxParameterCount="1000"/>
```
to this 
```
    <Connector port="80" protocol="HTTP/1.1"
               connectionTimeout="20000"
               maxParameterCount="1000"/>
```

**In /opt/tomcat/conf/web.xml**
I commented the following section like this 
```
    <!-- Force HTTPS, required for HTTP redirect -->
    <!--
    <security-constraint>
        <web-resource-collection>
            <web-resource-name>Protected Context</web-resource-name>
            <url-pattern>/*</url-pattern>
        </web-resource-collection>
       <user-data-constraint>
            <transport-guarantee>CONFIDENTIAL</transport-guarantee>
       </user-data-constraint>
    </security-constraint>
    -->
```

Conclusion : 
After these changes, TOMCAT stopped redirecting HTTP requests to port 443.
And now, my SMP can be queried correctly on port 80 with HTTP protocol !

**Reply by phax** [reply]
Thanks for sharing @Robcio35  - highly appreciated!
